### PR TITLE
Bah 10985

### PR DESCRIPTION
**Pull request by elder-joseph**
## Description
As a Comparison Tool user, I need the search functionality to be smarter so that results are found when I have small typos.
This is the `gibct-data-service` specific pull request for [10985](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/10985)

## Testing done
`ci` Testing passes locally, additional unit tests added for institution programs where functionality was added.

## Screenshots
N/A

## Acceptance criteria
- [x] The free text search on the VETTEC search results page label
"City, VET TEC program or provider"
is replaced with
"Program, provider, or location"

- [x]The 'physical institution zip code' from the WEAMS data is searchable within the Comparison Tool for VETTEC providers.

- [x]VETTEC search results return matches that partially match the following search criteria:
a. City
b. Institution Name

- [x] The institution search results are sorted in the following order:
a. % Match
b. Provider Name

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (zurbergram) [inline yarn.lock:1]: where did this come from?

File: app/models/institution_program.rb [hunk 2]
```diff
@@ -44,16 +44,23 @@ def va_bah
...
       'institution_programs.facility_code = (:facility_code)',
-      'institutions.institution LIKE (:upper_search_term)',
-      'lower(description) LIKE (:search_term)',
-      'lower(institutions.physical_city) LIKE (:search_term)'
+      'lower(description) LIKE (:search_term)'
     ]
 
+    if fuzzy_search
+      clause << 'institution % :upper_search_term'
+      clause << 'physical_city % :upper_search_term'
```
Comment (zurbergram) [inline app/models/institution_program.rb:57]: the `%` operator is most likely not going to work, see https://github.com/department-of-veterans-affairs/gibct-data-service/pull/685 for solution

File: yarn.lock (hunk 1 above)
Comment (elder-joseph) [reply yarn.lock:1]: That needs to go, will drop that

File: app/models/institution_program.rb (hunk 2 above)
Comment (elder-joseph) [reply app/models/institution_program.rb:57]: I can update that, was basing it off of what was there before #685

Comment (elder-joseph) [reply app/models/institution_program.rb]: @zurbergram pushed a change here to use `SIMILARITY` over the wildcards

**Review by zurbergram — APPROVED** [review]

**Review by sonntag-adam — APPROVED** [review]

**Review by omgitsbillryan — APPROVED** [review]

**Pull request merged**